pkg/registry/cache.go

```diff
@@ -2,13 +2,105 @@ package registry
 
 import (
 	"context"
+	"fmt"
 	"sync"
 	"time"
 
+	"github.com/adevinta/noe/pkg/log"
 	"github.com/prometheus/client_golang/prometheus"
 	"sigs.k8s.io/controller-runtime/pkg/metrics"
 )
 
+type CacheEntry[T any] struct {
+	Value  *T
+	Expiry time.Time
+}
+
+type Cache[T any] struct {
```

Comment:
Is it required to make Cache accept a type if it's not used in any field? It seems the only structure requiring generics is CacheEntry. I haven't played enough with generics yet, so is this a requirement?

Reply:
`T` conditions the value returned by [Load](https://github.com/adevinta/noe/pull/98/files#diff-0286a019339075435d8d4fb4abd4f45b3cdb69b80aac2d11367331ad0256f6f2R34) and [LoadOrCall](https://github.com/adevinta/noe/pull/98/files#diff-0286a019339075435d8d4fb4abd4f45b3cdb69b80aac2d11367331ad0256f6f2R44)

Not using generics here would impose the cache client to perform the [relevant cast](https://github.com/adevinta/noe/pull/98/files#diff-0286a019339075435d8d4fb4abd4f45b3cdb69b80aac2d11367331ad0256f6f2R36)